enumflags/src/lib.rs

```diff
@@ -204,6 +208,22 @@ where
         }
     }
 
+    pub fn from_flag(t: T) -> Self {
+        BitFlags { val: t.bits() }
+    }
+
+    pub fn try_from_bits(bits: T::Type) -> Result<Self, FromBitsError<T>> {
+        let flags = Self::from_bits_truncate(bits);
+        if flags.bits() == bits {
+            Ok(flags)
+        } else {
+            Err(FromBitsError {
+                flags,
+                invalid: bits & !flags.bits(),
+            })
+        }
+    }
+
```

Comment:
I don't think these should be also exposed as a method. A trait would be enough. In fact, I'm considering #[deprecating] `from_bits` once this lands.